pkg/services/store/k8saccess/controller.go

```diff
@@ -0,0 +1,57 @@
+package k8saccess
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/apimachinery/bridge"
+	"github.com/grafana/grafana/pkg/registry/corecrd"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/tools/cache"
+)
+
+type DashboardController struct {
+	bridgeService *bridge.Service
+	reg           *corecrd.Registry
+}
+
+func ProvideDashboardController(bridgeService *bridge.Service, reg *corecrd.Registry) *DashboardController {
+	return &DashboardController{
+		bridgeService: bridgeService,
+		reg:           reg,
+	}
+}
+
+func (c *DashboardController) Run(ctx context.Context) error {
+	dashboardCRD := c.reg.Dashboard()
+
+	gvr := schema.GroupVersionResource{
+		Group:    dashboardCRD.GVK().Group,
+		Version:  dashboardCRD.GVK().Version,
+		Resource: dashboardCRD.Schema.Name,
+	}
+
+	factory := dynamicinformer.NewDynamicSharedInformerFactory(c.bridgeService.ClientSet.Dynamic, time.Minute)
+	dashboardInformer := factory.ForResource(gvr).Informer()
+
+	dashboardInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			fmt.Printf("dashboard added: %s \n", obj)
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			fmt.Printf("dashboard changed: %s \n", newObj)
+		},
+		DeleteFunc: func(obj interface{}) {
+			fmt.Printf("dashboard deleted: %s \n", obj)
+		},
```

Comment:
Totally understand that for PoC we're handwriting these.

i just want to take the opportunity of this code existing (🎉🎉!!) to note what's done here is something we should iterate towards doing fully generically, combining codegen, conventional logic and configuration-driven behavior.

Reply:
totally 💯 💯 💯  - this, right now, is where we need to fudge and futz by hand in order to figure out what the consistent generic approach should look like